x/tokenomics/keeper/keeper_test.go

```diff
@@ -0,0 +1,136 @@
+	ctx     context.Context
+	keepers keepertest.TokenomicsModuleKeepers
+	claim   prooftypes.Claim
+}
+
+func (s *TestSuite) SetupTest() {
+	supplierAddr := sample.AccAddress()
+	appAddr := sample.AccAddress()
+
+	// Prepare and insert the claim
+	s.claim = prooftypes.Claim{
+		SupplierAddress: supplierAddr,
+		SessionHeader: &sessiontypes.SessionHeader{
+			ApplicationAddress:      appAddr,
+			Service:                 &sharedtypes.Service{Id: testServiceId},
+			SessionId:               "session_id",
+			SessionStartBlockHeight: 1,
+			SessionEndBlockHeight:   sessionkeeper.GetSessionEndBlockHeight(1),
+		},
+		RootHash: []byte("default_roo_hash"),
+	}
+
+	s.keepers, s.ctx = keepertest.NewTokenomicsModuleKeepers(s.T())
+	s.sdkCtx = sdk.UnwrapSDKContext(s.ctx)
+}
+
+func TestSettleExpiringSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (s *TestSuite) TestClaimSettlesWhenAProofIsRequired() {
+	// Retrieve default values
+	t := s.T()
+	ctx := s.ctx
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	claim := s.claim
+
+	// Add the claim & verify it exists
+	s.keepers.UpsertClaim(ctx, claim)
+	claims := s.keepers.GetAllClaims(ctx)
+	s.Require().Len(claims, 1)
+
+	// Settle expiring claims at height 2 (while the session is still active).
+	// Expectations: No claims should be settled because the session is still ongoing
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight - 2)
+	numClaimsSettled, numClaimsExpired, err := s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claim still exists
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 1)
+
+	// Try to settle expiring claim a little after it ended.
+	// Goal: Claims should not be settled because the proof window hasn't closed yet.
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight + 2)
+	numClaimsSettled, numClaimsExpired, err = s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claim still exists
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 1)
+
+	// Try to settle expiring claims a long time after it ended
+	// Expectation: All (1) claims should be settled.
+	sdkCtx = sdkCtx.WithBlockHeight(claim.SessionHeader.SessionEndBlockHeight * 10)
+	numClaimsSettled, numClaimsExpired, err = s.keepers.SettlePendingClaims(sdkCtx)
+	// Check that no claims were settled
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), numClaimsSettled)
+	require.Equal(t, uint64(0), numClaimsExpired)
+	// Validate that the claims expired
+	claims = s.keepers.GetAllClaims(ctx)
+	require.Len(t, claims, 0)
+
+	// Confirm an expiration event was emitted
+	events := sdkCtx.EventManager().Events()
```

Comment:
Are we not able to assert something more specific about the event or do you think it's just out of scope?